06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js

```diff
@@ -0,0 +1,353 @@
+const InvertedIndex = require('../public/builds/InvertedIndex').InvertedIndex;
+
+const books = require('../booksamples/books.json');
+const booksInvalidContent = require('../booksamples/booksInvalidContent.json');
+const bookBadJson = require('../booksamples/bookBadJson.json');
+const bookEmptyJson = require('../booksamples/bookEmptyJson.json');
+
+const bookText = ' ';
+
+const invertedIndex = new InvertedIndex();
+
+describe('iDex Inverted Index', () => {
+  beforeEach((done) => {
+    invertedIndex.createIndex('books', books);
+    invertedIndex.createIndex('booksInvalidContent', booksInvalidContent);
+    invertedIndex.createIndex('bookBadJson', bookBadJson);
+    invertedIndex.createIndex('bookText', bookText);
+    done();
+  });
+  describe('The InvertedIndex class', () => {
+    it('should be defined', () => {
+      expect(InvertedIndex).toBeDefined();
+    });
+
+    it('should have a defined instance', () => {
+      expect(invertedIndex).toBeDefined();
+      expect(invertedIndex instanceof InvertedIndex).toBeTruthy();
+    });
+
+    it('should default instance properties to object types', () => {
+      expect(invertedIndex.iDexMapper).toBeDefined();
+      expect(typeof invertedIndex.iDexMapper).toEqual('object');
+
+      expect(invertedIndex.numberOfDocuments).toBeDefined();
+      expect(typeof invertedIndex.numberOfDocuments).toEqual('object');
+
+      expect(invertedIndex.unIndexedBooks).toBeDefined();
+      expect(invertedIndex.unIndexedBooks).toEqual({});
+      expect(typeof invertedIndex.unIndexedBooks).toEqual('object');
+
+      expect(invertedIndex.indexedBookTitles).toBeDefined();
+      expect(typeof invertedIndex.indexedBookTitles).toEqual('object');
+    });
+  });
+
+  describe('The readFile method', () => {
+    it('should return the json file if it is valid', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(books))).toEqual((books));
+    });
+
+    it('should return the json file for an empty json', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(bookEmptyJson)))
+        .toEqual((bookEmptyJson));
+    });
+
+    it('should return false if the file cannot be parsed', () => {
+      expect(InvertedIndex.readFile((bookText))).toBe((false));
+      expect(InvertedIndex.readFile((bookText))).toBeFalsy();
+    });
+  });
+
+  describe('The validateFile method', () => {
```

Comment:
Generally the language used on your tests should be understood by both the developer and the customer. It helps you shape the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", () => {

    it("should be able to add 2 numbers together", () => {
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", () => {
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```
The result of this will read:
**_Calculator: add_**
    _should be able to add 2 numbers together_
   _should be able to add 3 numbers together_

Please review the language used for your validateFile test. Paying more attention to the language used on your "it" and 'describe' functions.